File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // Create HashMap 
        int[] ans = new int[2] ; 
        HashMap<Integer,Integer> map = new HashMap<>() ; 
        int n = nums.length ; 
        map.put(nums[0] , 0 ) ; 
        // travel on nums check req sum --> map 

        for ( int i = 1 ; i <  n ; i++ ) {
            int ele = nums[i]  ; 
            int reqSum = target - ele; 

            if ( map.containsKey(reqSum)) {
                int idx1 = map.get(reqSum) ; 
                int idx2 = i ; 

                ans[0] = idx1 ; 
                ans[1] = idx2 ; 

                break ; 
            }
            map.put(ele , i ) ; 
        }
        return ans ; 
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {} 
        L = [0,0]
        for i , ele in enumerate(nums) :
            if target - ele in dict :
                L[0] = i 
                L[1] = dict[target-ele] 
                return L
            dict[ele] = i 
        return L         